### PR TITLE
Authorization filter for node repository

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

**Comment by mpolden** [comment]
FYI @tokle

File: node-repository/pom.xml [hunk 1]
```diff
@@ -77,6 +77,11 @@
             <version>${project.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+            <scope>provided</scope>
```
Comment (bjorncs) [inline node-repository/pom.xml]: This must be compile scope - the bouncycastle bundles are not provided jdisc/configserver.

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/AuthorizerTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.yahoo.vespa.hosted.provision.restapi.v2;
```
Comment (bjorncs) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/AuthorizerTest.java]: Missing copyright header

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    AuthorizationFilter(Authorizer authorizer, BiConsumer<ErrorResponse, ResponseHandler> responseWriter) {
+        this.authorizer = authorizer;
+        this.responseWriter = responseWriter;
+    }
+
+    @Override
+    public void filter(DiscFilterRequest request, ResponseHandler handler) {
+        Optional<X509Certificate> cert = certificateFrom(request);
+        if (!cert.isPresent() || !authorizer.test(() -> commonName(cert.get()), request.getUri())) {
```
Comment (bjorncs) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/AuthorizationFilter.java]: The filter should distinguish 403 Forbidden (authorization failure) from 401 Unauthorized (authentication failure). Missing credentials => 401, invalid credentials => 403. See https://stackoverflow.com/a/6937030 :)

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/AuthorizationFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
+package com.yahoo.vespa.hosted.provision.restapi.v2.filter;
```
Comment (bjorncs) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/AuthorizationFilterTest.java]: Missing copyright header

**Review by bjorncs — CHANGES_REQUESTED** [review]

**Comment by mpolden** [comment]
@bjorncs PTAL.

**Review by bjorncs — APPROVED** [review]

**Comment by mpolden** [comment]
Hold merge please, found a few more cases that needs to be handled.

**Pull request merged**